flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/GenericCatalogTable.java

```diff
@@ -33,30 +35,53 @@
 public class GenericCatalogTable implements CatalogTable {
 	// Schema of the table (column names and types)
 	private final TableSchema tableSchema;
-	// Properties of the table
-	private final Map<String, String> properties;
 	// Statistics of the table
 	private final TableStats tableStats;
+	// Partition keys if this is a partitioned table. It's an empty set if the table is not partitioned
+	private final Set<String> partitionKeys;
```

Comment:
Should we use list to preserve the order ? IIUC, hive has orders when there're multiple partition columns.